test/add.js

```diff
@@ -38,222 +38,268 @@ function makeFileWithInlineSourceMap() {
-		t.ok(data instanceof File, 'should pass a vinyl file through');
-		t.ok(data.sourceMap, 'should add a source map object');
-		t.equal(String(data.sourceMap.version), '3', 'should have version 3');
-		t.deepEqual(data.sourceMap.sources, ['test1.js', 'test2.js'], 'should have right sources');
-		t.deepEqual(data.sourceMap.sourcesContent, ['console.log(\'line 1.1\');\nconsole.log(\'line 1.2\');\n', 'console.log(\'line 2.1\');\nconsole.log(\'line 2.2\');'], 'should have right sourcesContent');
-		t.equal(data.sourceMap.mappings, 'AAAAA,QAAAC,IAAA,YACAD,QAAAC,IAAA,YCDAD,QAAAC,IAAA,YACAD,QAAAC,IAAA', 'should have right mappings');
-		t.end();
-	});
-});
+		it('should not accept empty string as options argument', function(done) {
+			var file = makeFile();
+			[
+				{ type: 'a string', val: '' }
+			].map(function(obj) {
+				sourcemaps.add(file, obj.val, function(err, file) {
+					expect(err instanceof Error && err.message === 'vinyl-sourcemap-add: Invalid argument: options').toExist();
+					done();
+				});
+			});
+		});
 
-test('add: should remove inline sourcemap', function (t) {
-	sourcemaps.add(makeFileWithInlineSourceMap(), { loadMaps: true }, function(err, data) {
-		t.notOk(/sourceMappingURL/.test(data.contents.toString()), 'should not have sourcemapping');
-		t.end();
-	});
-});
 
-test('add: should load external source map file referenced in comment with the \/\/# syntax', function (t) {
-	var file = makeFile();
-	file.contents = new Buffer(sourceContent + '\n//# sourceMappingURL=helloworld2.js.map');
-	sourcemaps.add(file, { loadMaps: true }, function(err, data) {
-		t.ok(data.sourceMap, 'should add a source map object');
-		t.equal(String(data.sourceMap.version), '3', 'should have version 3');
-		t.deepEqual(data.sourceMap.sources, ['helloworld2.js'], 'should have right sources');
-		t.deepEqual(data.sourceMap.sourcesContent, ['source content from source map'], 'should have right sourcesContent');
-		t.equal(data.sourceMap.mappings, '', 'should have right mappings');
-		t.end();
-	});
-});
+		it('should not accept boolean as options argument', function(done) {
+			var file = makeFile();
+			[
+				{ type: 'a boolean', val: true }
+			].map(function(obj) {
+				sourcemaps.add(file, obj.val, function(err, file) {
+					expect(err instanceof Error && err.message === 'vinyl-sourcemap-add: Invalid argument: options').toExist();
+					done();
+				});
+			});
+		});
 
-test('add: should remove source map comment with the \/\/# syntax', function (t) {
-	var file = makeFile();
-	file.contents = new Buffer(sourceContent + '\n//# sourceMappingURL=helloworld2.js.map');
-	sourcemaps.add(file, { loadMaps: true }, function(err, data) {
-		t.notOk(/sourceMappingURL/.test(data.contents.toString()), 'should not have sourcemapping');
-		t.end();
-	});
-});
+		it('should add an empty sourceMap', function(done) {
+			sourcemaps.add(makeFile(), function(err, data) {
+				expect(File.isVinyl(data)).toExist(['Should return a vinyl file']);
```

Comment:
^ should be fixed throughout.  I usually don't specify an extra error message because `expect` makes easy-to-read assertions